dbms/src/Storages/Transaction/RegionPersister.cpp

```diff
@@ -211,20 +213,79 @@ RegionMap RegionPersister::restore(const TiFlashRaftProxyHelper * proxy_helper,
                     c,
                     provider);
             }
+            break;
         }
+        case PageStorageRunMode::ONLY_V3:
+        {
+            mergeConfigFromSettings(global_context.getSettingsRef(), config);
+
+            auto page_storage_v3 = std::make_shared<PS::V3::PageStorageImpl>( //
+                "RegionPersister",
+                path_pool.getPSDiskDelegatorGlobalMulti("kvstore"),
+                config,
+                provider);
+            page_storage_v3->restore();
+            page_writer = std::make_shared<PageWriter>(run_mode, /*storage_v2_*/ nullptr, page_storage_v3);
+            page_reader = std::make_shared<PageReader>(run_mode, ns_id, /*storage_v2_*/ nullptr, page_storage_v3, global_context.getReadLimiter());
+            break;
+        }
+        case PageStorageRunMode::MIX_MODE:
+        {
+            auto page_storage_v2 = std::make_shared<PS::V2::PageStorage>(
+                "RegionPersister",
+                delegator,
+                config,
+                provider);
+            // V3 should not used getPSDiskDelegatorRaft
+            // Because V2 will delete all invalid(unrecognized) file when it restore
+            auto page_storage_v3 = std::make_shared<PS::V3::PageStorageImpl>( //
+                "RegionPersister",
+                path_pool.getPSDiskDelegatorGlobalMulti("kvstore"),
+                config,
+                provider);
```

Comment:
Note that we use
* `$DATA_DIR/kvstore` for kvstore (PageStorage V2/V1)
* `$DATA_DIR/page/kvstore` for kvstore (PageStorage V3)

/cc @lidezhu @flowbehappy

Reply:
> Note that we use
> 
> * `$DATA_DIR/kvstore` for kvstore (PageStorage V2/V1)
> * `$DATA_DIR/page/kvstore` for kvstore (PageStorage V3)
> 
> /cc @lidezhu @flowbehappy

PS: Because V2 will delete invalid(unrecognized) files in `restore`.(So we can't use the same path for both V2 and V3. Otherwise, V3 files will be removed after V2 restored.